### PR TITLE
[O2B-1339] API for setting skimming productions

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- NA

Notable changes for developers:
- Added API for setting skimmable data pass

Changes made to the database:
- NA

### CODE REVIEW

File: lib/server/controllers/dataPasses.controller.js [hunk 1]
```diff
@@ -55,6 +55,27 @@ const listDataPassesHandler = async (req, res) => {
     }
 };
 
+// eslint-disable-next-line valid-jsdoc
+/**
+ * Set given data pass (for PROTON_PROTON runs) as skimmable
+ */
+const setAsSkimmableHandler = async (req, res) => {
```
Comment (martinboulais) [inline lib/server/controllers/dataPasses.controller.js:62]: markAs instead of setAs no?

File: lib/server/services/dataPasses/DataPassService.js [hunk 2]
```diff
@@ -139,6 +142,37 @@ class DataPassService {
...
+            const dataPass = await this.getOneOrFail(identifier);
+            if (dataPass.pdpBeamType !== PdpBeamType.PROTON_PROTON) {
+                throw new BadParameterError(`Cannot mark ${dataPass.name} as skimmable.`
+                    + ' Only production for PROTON_PROTON runs can be marked as skimmable');
+            }
+            if (!validSkimmableProductionNameRegex.test(dataPass.name)) {
+                throw new BadParameterError(`Cannot mark ${dataPass.name} as skimmable. Only \`apass\` can be marked as skimmable`);
+            }
+
+            const dataPassDB = await DataPassRepository.findOne({ where: { id: dataPass.id } });
```
Comment (martinboulais) [inline lib/server/services/dataPasses/DataPassService.js:164]: Why DB suffix?

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/server/controllers/dataPasses.controller.js (hunk 1 above)
Comment (xsalonx) [reply lib/server/controllers/dataPasses.controller.js:62]: applied

File: lib/server/services/dataPasses/DataPassService.js (hunk 2 above)
Comment (xsalonx) [reply lib/server/services/dataPasses/DataPassService.js:164]: Convention like in MonAlisa Synchronizer, to indicate it's sequelize model instance, but not an entity

**Pull request merged**